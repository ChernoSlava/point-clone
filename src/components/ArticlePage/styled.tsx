import styled from 'styled-components';

const ArticlePageStyled = styled.div`
    height: 100%;
    max-width: 984px;
    padding: 24px;
    border-radius: 8px;
`;

const ArticleTitleStyled = styled.h2`
    padding: 0;
    margin: 0;
    color: rgb(15, 23, 42);
    font-size: 40px;
    letter-spacing: -1px;
    line-height: 44px;
    font-weight: 700;
    margin: 12px 0px 30px;
    text-align: center;
`;

const ArticleDescriptionStyled = styled.p`
    padding: 0;
    margin: 0;
    color: rgb(15, 23, 42);
    font-size: 20px;
    font-weight: 400;
    letter-spacing: 0px;
    margin-top: 16px;
    margin-bottom: 24px;
    line-height: 28px;
`;
const ArticleImageStyled = styled.img`
    max-width: 100%;
    border-radius: 10px;
`;

const ArticleBlotStyled = styled.p`
    padding: 0;
    margin: 0;
    margin: 20px 0;
`;

export {
    ArticlePageStyled,
    ArticleTitleStyled,
    ArticleDescriptionStyled,
    ArticleImageStyled,
    ArticleBlotStyled,
}
