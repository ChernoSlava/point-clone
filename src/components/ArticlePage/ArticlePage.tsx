import React from 'react';
import { useLocation } from 'react-router-dom';
import { ArticlePageStyled, ArticleTitle, ArticleDescription } from './styled';
import { BackButton } from '../BackButton';

const ArticlePage: React.FC = () => {
  const location = useLocation();
  const articleTitle = location.state.articleTitle;

  console.log(articleTitle);
  return (
    <ArticlePageStyled>
      <ArticleTitle>{articleTitle}</ArticleTitle>
      <ArticleDescription>Шикарные новости для вас :D</ArticleDescription>
      <BackButton />
    </ArticlePageStyled>
  );
};

export { ArticlePage };
