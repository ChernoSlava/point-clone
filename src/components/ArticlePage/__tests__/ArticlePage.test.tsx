import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";

import { ArticlePage } from "../ArticlePage";

describe("ArticlePage tests", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("renders with correct title, image, and description", () => {
    const title = "Test Title";
    const image = "test-image.jpg";
    const description = "Test Description";

    const { container, unmount } = render(
      <MemoryRouter>
        <ArticlePage title={title} image={image} description={description} />
      </MemoryRouter>
    );

    expect(screen.getByText(title)).toBeTruthy();
    expect(screen.getByAltText("Big ico")).toBeTruthy();
    expect(screen.getByText(description)).toBeTruthy();

    expect(container).toMatchSnapshot();
    unmount();
  });

  test("renders BackButton", () => {
    const { container, unmount } = render(
      <MemoryRouter>
        <ArticlePage
          title="Test Title"
          image="test-image.jpg"
          description="Test Description"
        />
      </MemoryRouter>
    );

    expect(screen.getByText("Назад")).toBeTruthy();
    expect(container).toMatchSnapshot();
    unmount();
  });
});
