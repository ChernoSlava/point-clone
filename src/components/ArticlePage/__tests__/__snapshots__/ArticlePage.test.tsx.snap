// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePage tests renders BackButton 1`] = `
<div>
  <div
    class="sc-beySPh dWkjbF"
  >
    <h2
      class="sc-guDLey cygvQJ"
    >
      Test Title
    </h2>
    <p
      class="sc-dmyCSP dHLvlV"
    >
      Test Description
    </p>
    <img
      alt="Big ico"
      class="sc-hLQSwg bcGlma"
      src="test-image.jpg"
    />
    <p
      class="sc-eDLKkx flLHu"
    >
      Шикарные новости для вас :D
    </p>
    <a
      href="/"
      style="text-decoration: none; color: blue;"
    >
      Назад
    </a>
  </div>
</div>
`;

exports[`ArticlePage tests renders with correct title, image, and description 1`] = `
<div>
  <div
    class="sc-beySPh dWkjbF"
  >
    <h2
      class="sc-guDLey cygvQJ"
    >
      Test Title
    </h2>
    <p
      class="sc-dmyCSP dHLvlV"
    >
      Test Description
    </p>
    <img
      alt="Big ico"
      class="sc-hLQSwg bcGlma"
      src="test-image.jpg"
    />
    <p
      class="sc-eDLKkx flLHu"
    >
      Шикарные новости для вас :D
    </p>
    <a
      href="/"
      style="text-decoration: none; color: blue;"
    >
      Назад
    </a>
  </div>
</div>
`;
